File: tests/Modify.ts
import * as anchor from "@project-serum/anchor";import { Program } from "@project-serum/anchor";
import { UnsafePassword } from "../target/types/unsafe_password";
const provider = anchor.AnchorProvider.env();anchor.setProvider(provider);
const wallet = provider.wallet as anchor.Wallet;
const account = new anchor.web3.PublicKey("5rLPJrgrNyWZ3TwvbcJvGmV5qaJdBZ9c72dG3cCbywzW");
describe("test", () => {
    const program = anchor.workspace.UnsafePassword as Program<UnsafePassword>;
    it("Is initialized!", async () => {
      const tx = await program.methods.modify(
        "Lovelace",// instagram 
        "Gates",// github
        "Gosling",//facebook 
        "James",// gmail 
        "Rossum",// discord
        "Turing",// phamtom
        "Ritchie",// metamask
        "Hejlsberj",// solsea
        "Berners-Lee",// twitter
      ).accounts({warmHold: account,user: provider.wallet.publicKey}).signers([wallet.payer]).rpc();
      console.log("Your transaction signature", tx);
      console.log("Modified Account: ", account.toBase58()); 
    });
})

File: src/functions/modify.rs
use anchor_client::{
    solana_sdk::{
        pubkey::Pubkey,
        signature::Signature
    },
    Program,
};
use anyhow::Result;
use unsafe_password::Unsafehold;

pub fn modify(
    program: &Program,
    warm_hold: Pubkey,
    instagram: String,
    github: String,
    facebook: String,
    gmail: String,
    discord: String,
    phamtom: String,
    metamask: String,
    solsea: String,
    twitter: String,
) -> Result<()> {
    let tx: Signature = program
        .request()
        .accounts(unsafe_password::accounts::Modify {
            warm_hold: warm_hold,
            user: program.payer(),
        })
        .args(unsafe_password::instruction::Modify {
            instagram,
            github,
            facebook,
            gmail,
            discord,
            phamtom,
            metamask,
            solsea,
            twitter,
        })
        .send()?;
    let account: Unsafehold = program.account(warm_hold)?;
    println!("Tx: {}", tx);
    println!("Account: {}", warm_hold);
    println!("instagram: {}", account.instagram);
    println!("github: {}", account.github);
    println!("facebook: {}", account.facebook);
    println!("gmail: {}", account.gmail);
    println!("discord: {}", account.discord);
    println!("phamtom: {}", account.phamtom);
    println!("metamask: {}", account.metamask);
    println!("solsea: {}", account.solsea);
    println!("twitter: {}", account.twitter);
    Ok(())
}
